test/manipulation.test.js

```diff
@@ -1097,6 +1098,71 @@ describe('manipulation', function() {
     });
   }
 
+  var hasReplaceById = !!getSchema().connector.replaceById;
+  bdd.describeIf(hasReplaceById, 'replaceOrCreate when forceId is true', function() {
+    var Post, foundId;
+    var ds = getSchema();
+    before(function(done) {
+      Post = ds.define('Post', {
+        title: {type: String, length: 255, index: true},
+        content: {type: String},
+        comments: [String],
+      }, {forceId: true});
+      ds.automigrate('Post', done);
+    });
+
+    it('fails when id does not exist in db', function(done) {
+      var post = {id: 123, title: 'a', content: 'AAA'};
+      Post.replaceOrCreate(post, function(err, p) {
+        err.statusCode.should.equal(404);
+        done();
+      });
+    });
+
+    it('works on create if the request does not include an ID', function(done) {
+      var post = {title: 'a', content: 'AAA'};
+      Post.replaceOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.should.not.have.property('_id');
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        Post.findById(p.id, function(err, p) {
+          if (err) return done(err);
+          foundId = p.id;
+          done();
+        });
+      });
+    });
+
+    it('works on update if the request includes an exisiting ID in db', function(done) {
+      var post = {id: foundId, title: 'b', content: 'BBBB'};
+      Post.replaceOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.id.should.equal(post.id);
+        p.should.not.have.property('_id');
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        done();
+      });
+    });
+
+    it('works without options on update', function(done) {
+      Post.create({title: 'a', content: 'AAA', comments: ['Comment1']},
+          function(err, post) {
+            if (err) return done(err);
+            post = post.toObject();
+            delete post.comments;
+            delete post.content;
+            post.title = 'b';
+            Post.replaceOrCreate(post, function(err, p) {
+              if (err) return done(err);
+              p.id.should.equal(post.id);
+              done();
+            });
+          });
+    });
+  });
+
```

Comment:
I think these added tests ☝️ are breaking the original previous tests probably because they are using the same model name called `Post`; you should either move your tests after the original tests [here](https://github.com/strongloop/loopback-datasource-juggler/blob/b530ec2a6571ef2ab16abffb6298f35c1d975691/test/manipulation.test.js#L1295) or you may consider changing the model name not to have test model conflicts; that's just my guess to fix the broken tests [here](https://cis-jenkins.swg-devops.com/job/ds/job/loopback-connector-mysql~master/250/console); you need to investigate this further.